__tests__/app.test.js

```diff
@@ -353,4 +353,132 @@ describe("All requests", () => {
+                        expect(article).toHaveProperty("article_id", expect.any(Number));
+                        expect(article).toHaveProperty("topic", expect.any(String));
+                        expect(article).toHaveProperty("created_at", expect.any(String));
+                        expect(article).toHaveProperty("votes", expect.any(Number));
+                        expect(article).toHaveProperty("article_img_url", expect.any(String));
+                        expect(article).toHaveProperty("comments_count", expect.any(String));
+                    });
+                });
+        });
+        test("400 will returned and when topic query has invalid datatype", () => {
+            return request(app)
+                .get("/api/articles?topic=999&sort_by=title&order=asc")
+                .expect(400)
+                .then(({ body }) => {
+                    const { msg } = body;
+                    expect(msg).toBe("Bad request");
+                });
+        });
+        test("400 will returned and when sort_by query has invalid datatype", () => {
+            return request(app)
+                .get("/api/articles?topic=topic&sort_by=4&order=asc")
+                .expect(400)
+                .then(({ body }) => {
+                    const { msg } = body;
+                    expect(msg).toBe("Bad request");
+                });
+        });
+
+        test("400 will returned and when nonsense order query", () => {
+            return request(app)
+                .get("/api/articles?topic=topic&sort_by=title&order=nonsense")
+                .expect(400)
+                .then(({ body }) => {
+                    const { msg } = body;
+                    expect(msg).toBe("Bad request");
+                });
+        });
+        test("400 will returned and when passed all query wrong", () => {
+            return request(app)
+                .get("/api/articles?topic=topic&sort_by=title&order=nonsense")
+                .expect(400)
+                .then(({ body }) => {
+                    const { msg } = body;
+                    expect(msg).toBe("Bad request");
+                });
+        });
+        test("404 will returned and when passed valid topic query, but no resource found", () => {
+            return request(app)
+                .get("/api/articles?topic=paper&sort_by=title&order=asc")
+                .expect(404)
+                .then(({ body }) => {
+                    const { msg } = body;
+                    expect(msg).toBe("Not found!");
+                });
+        });
+        test("404 will returned and when passed valid sort_id query, but no resource found", () => {
+            return request(app)
+                .get("/api/articles?topic=mitch&sort_by=body&order=asc")
+                .expect(404)
+                .then(({ body }) => {
+                    const { msg } = body;
+                    expect(msg).toBe("Not found!");
+                });
+        });
```

Comment:
a invalid sort_by column is actually a bad path request, because the user is specifically limited on what they can sort_by (only the columns of the table) so it's more similar to when they use order incorrectly. this should be a 400